src/platform/K32W/Logging.cpp

```diff
@@ -21,26 +21,29 @@
 static bool isLogInitialized;
 extern uint8_t gOtLogUartInstance;
 extern "C" void K32WWriteBlocking(const uint8_t * aBuf, uint32_t len);
+extern "C" uint32_t otPlatAlarmMilliGetNow(void);
 
 namespace chip {
 namespace Logging {
 namespace Platform {
 
 void GetMessageString(char * buf, uint8_t chipCategory, uint8_t otLevelLog)
 {
+    sprintf(buf, "[%lu]", otPlatAlarmMilliGetNow());
+
     if (chipCategory != kLogCategory_None)
     {
         switch (chipCategory)
         {
         case kLogCategory_Error:
-            memcpy(buf, "[Error]", 7);
+            memcpy(buf + strlen(buf), "[Error]", 7);
```

Comment:
I don't understand how this was supposed to work before or how it works now.  buf isn't length delimited, but no null terminator is explicitly written.

Why not just use strcpy?  That would give you the null character and also alleviate the need to count chars.

Reply:
Seems like a strcat() scenario.

Reply:
@msandstedt  @jelderton , agree that the old code was messy. I refactored it using the snprintf function.